### PR TITLE
NFT updates

### PR DESCRIPTION
**Pull request by davidiw**
* Change ownership model so that owners get Tokens and the Collection has data
* Eliminate generics on almost all interfaces
* Have a dedicated siloed place for generic data to ease generic proliferation across the code

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/Token.move [hunk 1]
```diff
@@ -157,20 +146,51 @@ module AptosFramework::Token {
...
+    // Some Tokens may want additional fields outside the specification. As the Token itself does
+    // not contain the data, it is easier in Move to have this extra metadata sit on the side so
+    // that only creation and read operations (outside of Move) need to deal with the nuance of 
+    // metadata.
+    struct TokenMetadata<TokenType: store> has key {
+        metadata: Table<ID, TokenType>,
     }
 
-    public fun create_token_script<TokenType: copy + drop + store>(
+    fun initialize_token_metadata<TokenType: store>(account: &signer) {
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/Token.move:157]: how do we look up the data given a token? there's no address for the metadata specified in the token. (same for collections). probably need to attach creator address

**Review by zekun000 — COMMENTED** [review]
I don't have strong opinion on whether to store T aside or not. Note without generics, it's possible to store different tokens in the same collection/gallery. It's probably a feature for gallery, but it's weird for collection.

File: aptos-move/framework/aptos-framework/sources/Token.move [hunk 2]
```diff
@@ -107,44 +107,35 @@ module AptosFramework::Token {
     }
 
     // An account's set of Tokens
-    struct Gallery<TokenType: copy + drop + store> has key {
+    struct Gallery<phantom TokenType: copy + drop + store> has key {
         gallery: Table<ID, Token<TokenType>>,
     }
 
-    fun initialize_gallery<TokenType: copy + drop + store>(signer: &signer) {
+    fun initialize_gallery<TokenType: copy+ drop + store>(signer: &signer) {
```
Comment (msmouse) [inline aptos-move/framework/aptos-framework/sources/Token.move]: space after "copy"

File: aptos-move/framework/aptos-framework/sources/Token.move [hunk 3]
```diff
@@ -157,20 +146,51 @@ module AptosFramework::Token {
         uri: ASCII::String,
     }
 
-    public fun token_id<TokenType: copy + drop + store>(token: &Token<TokenType>): &ID {
-        &token.id
+    // Some Tokens may want additional fields outside the specification. As the Token itself does
+    // not contain the data, it is easier in Move to have this extra metadata sit on the side so
+    // that only creation and read operations (outside of Move) need to deal with the nuance of 
+    // metadata.
+    struct TokenMetadata<TokenType: store> has key {
```
[the remark this answers is not in the document]
Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/Token.move:153]: Subdata or what not, but the TokenData is what's in the collection and the ancillary data is elsewhere

File: aptos-move/framework/aptos-framework/sources/Token.move (hunk 1 above)
Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/Token.move:157]: The ID contains the creators address

**Review by zekun000 — APPROVED** [review]

**Pull request closed without merging**